pyvista/core/datasetattributes.py

```diff
@@ -788,7 +788,10 @@ def _prepare_array(
                 raise ValueError(
                     f'data length of ({data.shape[0]}) != required length ({array_len})',
                 )
-
+            if array_len > 0 and data.size == 0:
+                raise ValueError(
+                    f"Empty arrays are not allowed. Array '{name}' cannot have a size of 0."
```

Comment:
Maybe put `{data.shape}` in the error message too? I could see this message being a little confusing if the offending line is something like `mesh['foo'] = np.zeros((mesh.n_points, n_dims))` if `n_dims` has been set to zero a non-trivial number of lines of code away.